File: client/src/api/tag.ts
import http from './http'

// 標籤類型定義
export interface Tag {
  id: number
  tagname: string
  createtime?: string
  updatetime?: string
  isdelete?: number
}

// 獲取所有標籤
export function getAllTags() {
  return http.get('/tag/list')
}

// 更新用户標籤 - 適配新接口
export function updateUserTags(id: number, tags: string[]) {
  return http.post('/user/updateTags', {
    id,
    tags
  })
}

// 導出默認對象
export default {
  getAllTags,
  updateUserTags
} 

File: server/src/main/java/com/friendsync/stevenpang/model/Tag.java
package com.friendsync.stevenpang.model;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.util.Date;
import lombok.Data;

/**
 * 标签
 * @TableName tag
 */
@TableName(value ="tag")
@Data
public class Tag {
    /**
     * id
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 标签名称
     */
    private String tagname;

    /**
     * 创建时间
     */
    private Date createtime;

    /**
     * 
     */
    private Date updatetime;

    /**
     * 是否删除
     */
    private Integer isdelete;
}